pkg/querier/queryrange/queryrangebase/instrumentation.go

```diff
@@ -1,4 +1,4 @@
-package queryrange
+package queryrangebase
```

Comment:
hmmmm sounds better, indeed. I'll try renaming to `base`, then

Reply:
that's a big change (easy to make, but touches a lot), so I wonder if anyone else has opinions before you put in the effort?

Reply:
I'm going to be even more demanding, would it be possible to merge the two package in one ? This could help also remove the complexity of handling multiple type of requests in an another refactoring PR.

Unless we plan to re-vendor at some point ( we can talk about this offline) I think we should merge if this is possible.

Reply:
Where is the conflict with Loki's `queryrange` package? Is the issue that we also generate protobuf for this package in Loki? Otherwise I'd suggest to merge them.

Reply:
I think it's hard but not impossible. @DylanGuedes we can pair together if you want. Not going to block this PR though since it could be done after too.

Reply:
Actually, my first approach was to have them merged and yeah, a lot of things were conflicting. I decided to give up on that approach mostly because I considered it to be extremely hard to review but I can shoot it another try (in the future).

All that said, I vote for merging this as a separate package, and then I can merge it with the current Loki queryrange. The reason is, based on my experience merging this, it will take some precious time.

Reply:
Sounds good let's keep the merge of the package for another PR.

Reply:
I'll finish my review tomorrow then.